lib/rpi_auth/engine.rb

```diff
@@ -12,29 +13,40 @@ class Engine < ::Rails::Engine
       OmniAuth.config.logger = Rails.logger
     end
 
-    initializer 'RpiAuth.bypass_auth' do
-      RpiAuth.configuration.enable_auth_bypass
-    end
-
     initializer 'RpiAuth.add_middleware' do |app|
+      next unless RpiAuth.configuration
+
       app.middleware.use OmniAuth::Builder do
         provider(
-          OmniAuth::Strategies::Rpi,
-          RpiAuth.configuration.auth_client_id,
-          RpiAuth.configuration.auth_client_secret,
+          :openid_connect,
+          name: :rpi,
+          issuer: RpiAuth.configuration.issuer,
           scope: RpiAuth.configuration.scope,
           callback_path: '/rpi_auth/auth/callback',
+          response_type: RpiAuth.configuration.response_type,
+          client_auth_method: RpiAuth.configuration.client_auth_method,
           client_options: {
-            site: RpiAuth.configuration.auth_url,
-            authorize_url: "#{RpiAuth.configuration.auth_url}/oauth2/auth",
-            token_url: "#{RpiAuth.configuration.auth_token_url || RpiAuth.configuration.auth_url}/oauth2/token"
+            identifier: RpiAuth.configuration.auth_client_id,
+            secret: RpiAuth.configuration.auth_client_secret,
+            scheme: RpiAuth.configuration.token_endpoint.scheme,
+            host: RpiAuth.configuration.token_endpoint.host,
+            port: RpiAuth.configuration.token_endpoint.port,
+            authorization_endpoint: RpiAuth.configuration.authorization_endpoint,
+            token_endpoint: RpiAuth.configuration.token_endpoint,
+            jwks_uri: RpiAuth.configuration.jwks_uri
           },
-          authorize_params: { brand: RpiAuth.configuration.brand },
+          extra_authorize_params: { brand: RpiAuth.configuration.brand },
```

Comment:
I'm assuming we'd also need to add in support for the `login_options` param here (ref: https://github.com/RaspberryPiFoundation/omniauth-rpi/blob/5e54da89867917ea6eb9241a13a8f71060795ac2/lib/omniauth/strategies/hydra1.rb#L15-L21)

Reply:
I've updated to allow the `login_options`, and added a `rpi_auth_signup` route too, so that it is easy to use.